chainservice/chainservice.go

```diff
@@ -385,6 +385,9 @@ func New(
 			return nil, err
 		}
 	}
+	if dao == nil {
+		return nil, errors.New("blockdao is nil")
+	}
```

Comment:
why need this?

for test, we may have NewBlockDAO() return nil
for actual case like "make run", it should return a non-nil blockdao?